bot/exts/fun/quack.py

```diff
@@ -0,0 +1,62 @@
+from typing import Optional
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+API_URL = 'https://quackstack.pythondiscord.com'
+
+
+class Quack(commands.Cog):
+    """Cog for the .quack command."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    @commands.command()
+    async def quack(self, ctx: commands.Context, ducktype: Optional[str] = "duck", seed: Optional[str] = None) -> None:
+        """
+        Uses the Quackstack API to generate a random duck.
+
+        If a seed is provided, a duck is generated based on the given seed.
+        If a duck type is provided, either a duck or manduck will be generated.
+        """
+        ducktype = ducktype.lower()
+        if ducktype not in ["duck", "manduck"]:
+            raise commands.BadArgument(
+                "The duck type can only be `duck` or `manduck`."
+            )
+
+        quackstack_url = f"{API_URL}/{ducktype}"
+        params = {}
+        if seed is not None:
+            try:
+                seed = int(seed)
+            except ValueError:
+                newseed = ""
+                for e in seed:
+                    newseed += str(ord(e))
+                seed = int(newseed)
```

Comment:
I don't understand this, why the intermediate `str(ord())` stuff?

Reply:
Just in case the user submits a string as a seed

Reply:
Right, and how does work and why did you do this rather than something else like skipping the `str()` step.

Or is this just the first-best way to get a number from a string with characters?

Reply:
Just the best way

Reply:
Yeah so the implementation doesn't matter, can you add a comment for that for future uses?